c/array/rotate_array.c

```diff
@@ -1,29 +1,119 @@
-#include <array_utils.h>
+
+#include <stdbool.h>
+#include <assert.h>
+#include <time.h>
+#include "array_utils.h"
 
 #define MAX_SIZE 10
 
+/*
+ *rotate array rotaates the given original array for the given
+ *number of times.
+ *for example 1 2 3 4 5 6 7 8 9 10 is the original array
+ *then it roates it to 9 10 1 2 3 4 5 6 7 8 when number of times
+ *rotate is 2.
+*/
+
+void rotate_array(int *nums, int size, int number_of_times_rotate)
+{
+    for (int count = 0; count < number_of_times_rotate; count++)
+    {
+        int last_element = nums[size - 1];
+        for (int i = size - 1; i > 0; i--)
+        {
+            nums[i] = nums[i - 1];
+        }
+        nums[0] = last_element;
+    }
+}
+
+
+
+/* 
+ * is_array_rotated checks if the rotated array actually represents the
+ * origianal_array after nu number of rotations from the right hand side.
+*/
+bool is_array_rotated(int *rotated_array, int size, int n,int* orig_array)
+{
+
+
+// 1 2 3 4 5 6 7 8 9 10
+// n= 2
+
+//rotated_array
+
+//i=0; i<n; i++ 
+// j=size-1; j>size-n; j--
+// 9 10 1 2 3 4 5 6 7 8
+// 1 2 3 4 5 6 7 8 9 10
+
+    /* this loop checks the first n elements of the rotated array
+     * with the lat n elements of the original array in reverse orders.
+     * For example check the  9, 10 in rotatedarray with 10,9 in reverse.
+     */
+    for (int i=0, j=size-1; i<n; i++,j--)
+    {
+        if (rotated_array[i] != orig_array[j])
+        {
+            return false;
+        }
+    
+    }
+    /* this loop checks the first size-n elements of the rotated array
+     * with the first size-n elements of the original array in same order.
+     * For example check the  1 to 8 in rotatedarray with 1 to 8 in same order.
+     */
+    for (int i=n,j=0; i<size;i++,j++)
+    {
+        if (rotated_array[i] != orig_array[j])
+        {
+            return false;
+            
+        }
+    }
+
+    return true;
+}
+
+
+
 int main()
 {
     int number_of_times_rotate;
-    int my_nums[MAX_SIZE];
+    int* my_nums;
+    int* orig_array;
     srand(time(NULL));
-    
-    //int size = get_array_size_from_user();
-    //int *array = create_new_array_with_user_input(size);
 
-    // 1 2  3 4 5 6 7 8 9 10
-    // 9 10 1 2 3 4 5 6 7 8
+    number_of_times_rotate = rand() % MAX_SIZE; 
+    
+    /*
+    *This function call gets the user input for the original array.
```

Comment:
Inside a function, you can have comments with // for single line comments.

Reply:
No, change the comment to style like this:
//This function call gets the user input for the original array.